bedtools/2.31.1/Dockerfile

```diff
@@ -0,0 +1,83 @@
+ARG BEDTOOLS_VER="2.31.1"
+
+### builder stage for compiling bedtools code ###
+FROM ubuntu:jammy as builder
+
+# re-instantiate variable so we can use it in builder stage
+ARG BEDTOOLS_VER
+
+LABEL base.image="ubuntu:jammy"
+LABEL dockerfile.version="1"
+LABEL software="bedtools"
+LABEL software.version="${BEDTOOLS_VER}"
+LABEL description="bedtools - the swiss army knife for genome arithmetic"
+LABEL website="https://github.com/arq5x/bedtools2"
+LABEL license="https://github.com/arq5x/bedtools2/blob/master/LICENSE"
+LABEL maintainer="Curtis Kapsak"
+LABEL maintainer.email="kapsakcj@gmail.com"
+LABEL maintainer2="Erin Young"
+LABEL maintainer2.email="eriny@utah.gov"
+
+# install deps via apt-get, these are mainly for compiling bedtools code and for running tests. some are for downloading files (wget, ca-certificates)
+# last command is to point 'python' cmd to `python3` so that bedtools test scripts work. There are bash scripts that call 'python'
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \ 
+    zlib1g-dev \
+    libghc-bzlib-dev \
+    liblzma-dev \
+    wget \
+    ca-certificates \
+    python3 && \
+    apt-get autoclean && rm -rf /var/lib/apt/lists/* && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+# python3 required when compiling via `make` command for creating old CLI executables
+# dependencies listed here (albeit for v2.30.0, still should be identical): https://packages.ubuntu.com/jammy/bedtools
+# requires libghc-bzlib-dev, build-essential, zlib1g-dev, and a few others
+# 'make install' should place binary executable files in /usr/local/bin
+RUN wget -q https://github.com/arq5x/bedtools2/archive/refs/tags/v${BEDTOOLS_VER}.tar.gz && \
+   tar -xzf v${BEDTOOLS_VER}.tar.gz && \
+   cd bedtools2-${BEDTOOLS_VER} && \
+   make && \
+   make install 
+
+### keeping old installation cmds here in case we want to install via pre-compiled binary again in the future ###
+# RUN cd /usr/local/bin && \
+#     wget https://github.com/arq5x/bedtools2/releases/download/v${BEDTOOLS_VER}/bedtools.static && \
+#     mv bedtools.static bedtools && \
+#     chmod +x bedtools && \
+#     mkdir /data
+
+### test stage ###
+FROM builder as test
```

Comment:
"FROM app as test"? 
We need to test the app to see any missing libraries in the app stage. So, the test stage needs to be after the app stage.

Reply:
You will need to download the test files again in the test stage.

Reply:
bedtools has a lot of subcommands, but the internal tests of bedtools has a dependency from builder.

I think when we did this one before, we tested it with some files after.

Let me get this one up to standard practice, though.